Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -2302,10 +2304,20 @@ open class NavigationMapView: UIView {
      - parameter options: options, controlling caching parameters like area radius and concurrent downloading threads.
      */
     public func enablePredictiveCaching(options predictiveCacheOptions: PredictiveCacheOptions) {
-        let styleSourcePaths = mapView.styleSourceDatasets(["raster", "vector"])
-        
-        predictiveCacheManager = PredictiveCacheManager(predictiveCacheOptions: predictiveCacheOptions,
-                                                        styleSourcePaths: styleSourcePaths)
+        predictiveCacheMapObserver?.cancel()
+
+        self.predictiveCacheManager = PredictiveCacheManager(predictiveCacheOptions: predictiveCacheOptions,
+                                                             styleSourcePaths: styleSourceDatasetsForCaching())
+
+        self.predictiveCacheMapObserver = mapView.mapboxMap?.onEvery(event: .styleLoaded) { [weak self] _ in
```

Comment:
Parity with [Android](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt#L229)

```kotlin
        val onStyleLoadedListener = OnStyleLoadedListener {
        ...
            createMapsController(styleURI, map, tileStore)
```
Allows to cache styles that was not loaded at the moment of PredictiveCacheManager creation.

The observer is stored to correctly handle the case when method `enablePredictiveCaching` is called multiple times